internal/pkg/gateway/api_test.go

```diff
@@ -178,9 +203,33 @@ func TestGateway(t *testing.T) {
 				plan: endorsementPlan{
 					"g1": {"localhost:7051"},
 				},
-				signedProposal:       validSignedProposal,
-				processProposalError: fmt.Errorf("mumbo-jumbo"),
-				errString:            "mumbo-jumbo",
+				signedProposal: validSignedProposal,
+				endpointDefinition: &endpointDef{
+					proposalError: status.Error(codes.Aborted, "mumbo-jumbo"),
```

Comment:
The same error is being used in multiple places; this complicated my sniff test of some of your error logic earlier. It would be better to have unique strings.